rest-app/src/test/java/com/epam/repair/rest/OrdersDTORestControllerTest.java

```diff
@@ -0,0 +1,57 @@
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import static org.hamcrest.Matchers.hasSize;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+public class OrdersDTORestControllerTest {
+
+    @Autowired
+    private OrdersDTORestController ordersDTORestController;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    public void findAll() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.get("/orders")
+                .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.jsonPath("$", hasSize(2)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$").exists())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[*].ordersId").isNotEmpty())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[*].ordersStartDate").isNotEmpty())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[*].ordersEndDate").isNotEmpty())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[0].ordersId", Matchers.is(1)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[0].client.firstName.firstName", Matchers.is("Siarhei")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[0].client.lastName.lastName", Matchers.is("Arkhutsik")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[0].device.type.typeName", Matchers.is("smartphone")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[0].device.deviceCondition.appearance.appearanceName", Matchers.is("scratches")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[0].status.statusName", Matchers.is("In work")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[1].ordersId", Matchers.is(2)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[1].client.firstName.firstName", Matchers.is("Aliaksandr")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[1].client.lastName.lastName", Matchers.is("Ramaniuk")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[1].device.model.modelName", Matchers.is("xperia xz1")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[1].device.deviceCondition.complectation.complectationName", Matchers.is("case")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[1].status.statusName", Matchers.is("Waiting for a spare part")));
```

Comment:
A lot of checks which hard to read.
It would be great to use some other approach to check correctness of JSON.